### DISCUSSION TITLE
RIME與鼠鬚管的耦合方式究竟是C還是Cpp？

### DISCUSSION THREAD

**Discussion by ShikiSuen** [rime/home#1066] [Ideas]
我純屬好奇，尋思著要是鼠鬚管的所有 ObjC 內容用 Swift 重寫了的話會怎樣。
看到有個 utf8len 計算用的 Cpp 檔案，正好 Swift 內建了計算這種尺寸的功能（Swift 的 Strings 的 count() 會回傳 utf8length）。

剛好我也完成了威注音專案的徹底 Swift 化，自己親自重寫了語言模型引擎與注音 syllable composer。
於是在想要是鼠鬚管也 Swift 化的話會怎樣。（我一看 Cpp 就頭痛。）

副作用就是 app bundle 會帶一套 Swift 運行庫，我不知道佛振是否會對此不樂見。

**Comment by lotem** [comment]
介面是 C API
https://github.com/rime/librime/blob/master/src/rime_api.h

**Reply by ShikiSuen** [reply]
好消息。这样一来，鼠须管理论上可以全盘 Swift 化。
